sdk/identity/azure-identity-v2/src/main/java/com/azure/identity/v2/AzureToolkitCredential.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.v2;
+
+import com.azure.identity.v2.implementation.client.PublicClient;
+import com.azure.identity.v2.implementation.models.MsalToken;
+import com.azure.identity.v2.implementation.models.PublicClientOptions;
+import com.azure.identity.v2.implementation.util.IdentityConstants;
+import com.azure.identity.v2.implementation.util.LoggingUtil;
+import com.azure.v2.core.credentials.TokenCredential;
+import com.azure.v2.core.credentials.TokenRequestContext;
+import io.clientcore.core.credentials.oauth.AccessToken;
+import io.clientcore.core.instrumentation.logging.ClientLogger;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * <p>IntelliJ IDEA is an integrated development environment (IDE) developed by JetBrains, which provides a variety of
+ * features to support software development, such as code completion, debugging, and testing.
+ * Azure offers <a href="https://learn.microsoft.com/azure/developer/java/toolkit-for-intellij/">Azure Toolkit
+ * for IntelliJ plugin</a> for the IntelliJ IDEA development environment. It
+ * enables developers to create, test, and deploy Java applications to the Azure cloud platform. In order to
+ * use the plugin authentication as a user or service principal against
+ * <a href="https://learn.microsoft.com/entra/fundamentals/">Microsoft Entra ID</a> is required.
+ * The IntelliJCredential authenticates in a development environment and acquires a token on behalf of the
+ * logged-in account in Azure Toolkit for IntelliJ. It uses the logged in user information on the IntelliJ IDE and uses
+ * it to authenticate the application against Microsoft Entra ID.</p>
+ *
+ * <h2>Configure IntelliJCredential</h2>
```

Comment:
Generally if we are going to change the name we should maybe rework this doc a bit. Maybe the javadoc can have links out to setup instructions for each IDE?